daemon/logger/jsonfilelog/read.go

```diff
@@ -129,7 +134,23 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 	if err != nil {
 		logWatcher.Err <- err
 	}
-	defer fileWatcher.Close()
+	defer func() {
+		f.Close()
```

Comment:
Actually I should remove that one since `f` here can be a different fd due to rotation (and hence why I've put this in a closure).